contentctl/objects/abstract_security_content_objects/detection_abstract.py

```diff
@@ -338,171 +356,209 @@ def serialize_model(self):
+                f"DataSource objects:{missing_sources}"
+            )
+
         self.data_source_objects = matched_data_sources
 
         for story in self.tags.analytic_story:
-            story.detections.append(self)            
-        return self
+            # TODO (cmcginley): I think this instance of typing weirdness highlights some of the
+            #   issues I see with maintaining Detection and Detection_Abstract
+            story.detections.append(self)
+
+        # TODO (cmcginley): I'd still like to understand why the requested attr was nil when called
+        #   as part of CorrelationSearch, as well as when the "after" validators are called
+        self.cve_enrichment_func(__context)
 
-    
-    @field_validator('lookups',mode="before")
+    @field_validator('lookups', mode="before")
     @classmethod
-    def getDetectionLookups(cls, v:list[str], info:ValidationInfo)->list[Lookup]:
-        director:DirectorOutputDto = info.context.get("output_dto",None)
-        
-        search:Union[str,dict] = info.data.get("search",None)
-        if not isinstance(search,str):
-            #The search was sigma formatted (or failed other validation and was None), so we will not validate macros in it
+    def getDetectionLookups(cls, v: list[str], info: ValidationInfo) -> list[Lookup]:
+        if info.context is None:
+            raise ValueError("ValidationInfo.context unexpectedly null")
+
+        director: DirectorOutputDto = info.context.get("output_dto", None)
+
+        search: Union[str, dict] = info.data.get("search", None)                                    # type: ignore
+        if not isinstance(search, str):
+            # The search was sigma formatted (or failed other validation and was None), so we will
+            # not validate macros in it
             return []
-        
-        lookups= Lookup.get_lookups(search, director)
+
+        lookups = Lookup.get_lookups(search, director)
         return lookups
 
-    @field_validator('baselines',mode="before")
+    @field_validator('baselines', mode="before")
     @classmethod
-    def mapDetectionNamesToBaselineObjects(cls, v:list[str], info:ValidationInfo)->List[Baseline]:
+    def mapDetectionNamesToBaselineObjects(cls, v: list[str], info: ValidationInfo) -> List[Baseline]:
         if len(v) > 0:
-            raise ValueError("Error, baselines are constructed automatically at runtime.  Please do not include this field.")
+            raise ValueError(
+                "Error, baselines are constructed automatically at runtime.  Please do not include this field."
+            )
 
-        
-        name:Union[str,None] = info.data.get("name",None)
+        # TODO (cmcginley): if it's possible in practice for this to return None, we should adjust
+        #   the declared type
+        name: Union[str, dict] = info.data.get("name", None)                                        # type: ignore
```

Comment:
If it's possible in practice for this to return `None` that should be part of the type, otherwise the check on line 488 makes no sense.

Reply:
Yup, that's correct! I will make this change as well.